packages/bpk-component-drawer/src/BpkDrawer.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Portal } from 'bpk-react-utils';
+import { withScrim } from 'bpk-scrim-utils';
+
+import BpkDrawerContent from './BpkDrawerContent';
+
+const ScrimBpkDrawerContent = withScrim(BpkDrawerContent);
+
+class BpkDrawer extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      show: true,
```

Comment:
This should be provided by the component consumer rather than stored as state inside the component.

Reply:
No, it shouldn't. This state keeps track of the drawer being in view so animations trigger in order. Continued in next comment.